.github/workflows/cmake.yml

```
@@ -73,3 +73,125 @@ jobs:
+               --with-date_time --with-filesystem \
+               --with-system --with-iostreams \
+               --with-nowide --with-regex --with-test \
+               toolset=emscripten link=static variant=release threading=single install
+      - name: Build GMP
+        run: |
+          cd build
+          export GMP_VER=6.3.0
+          wget https://ftp.gnu.org/gnu/gmp/gmp-${GMP_VER}.tar.xz
+          tar -Jxf gmp-${GMP_VER}.tar.xz
+          mv gmp-${GMP_VER} gmp
+          cd gmp
+          emconfigure ./configure --disable-assembly --host none --enable-cxx --prefix=$(realpath dist)
+          emmake make
+          emmake make install
+      - name: Build MPFR
+        run: |
+          cd build
+          export MPFR_VER=4.2.1
+          wget https://ftp.gnu.org/gnu/mpfr/mpfr-${MPFR_VER}.tar.xz
+          tar -Jxf mpfr-${MPFR_VER}.tar.xz
+          mv mpfr-${MPFR_VER} mpfr
+          cd mpfr
+          emconfigure ./configure --host none --prefix=$(realpath dist) --with-gmp=$(realpath ../gmp/dist)
+          emmake make
+          emmake make install
+      - name: Build Ledger
+        env:
+          EXT: ${{ matrix.version.emsdk-ext }}
+        run: |
+          cd build
+          export BOOST="$(realpath boost/dist)"
+          # -lnodefs.js -lnoderawfs.js flags are for testing purposes.
+          emcmake cmake \
+            -DBUILD_LIBRARY=ON \
+            -DPRECOMPILE_SYSTEM_HH=OFF \
+            -DCMAKE_CXX_FLAGS=' -lnodefs.js -lnoderawfs.js ' \
+            -DCMAKE_INSTALL_PREFIX="$PWD"/dist \
+            -DBoost_ROOT="$BOOST" \
+            -DBoost_INCLUDE_DIR="$BOOST"/include \
+            -DBOOST_LIBRARYDIR="$BOOST"/lib \
+            -DBoost_DATE_TIME_LIBRARY_RELEASE="$BOOST"/lib/libboost_date_time.$EXT \
+            -DBoost_FILESYSTEM_LIBRARY_RELEASE="$BOOST"/lib/libboost_filesystem.$EXT \
+            -DBoost_IOSTREAMS_LIBRARY_RELEASE="$BOOST"/lib/libboost_iostreams.$EXT \
+            -DBoost_NOWIDE_LIBRARY_RELEASE="$BOOST"/lib/libboost_nowide.$EXT \
+            -DBoost_REGEX_LIBRARY_RELEASE="$BOOST"/lib/libboost_regex.$EXT \
+            -DBoost_SYSTEM_LIBRARY_RELEASE="$BOOST"/lib/libboost_system.$EXT \
+            -DGMP_PATH="$PWD"/gmp/dist/include \
+            -DGMP_LIB="$PWD"/gmp/dist/lib/libgmp.a \
+            -DMPFR_PATH="$PWD"/mpfr/dist/include \
+            -DMPFR_LIB="$PWD"/mpfr/dist/lib/libmpfr.a \
+            -DUTFCPP_INCLUDE_DIR="$PWD"/../lib/utfcpp/v4/source \
+            ..
+          emmake make -j4
+      - name: Fix Emscripten build
+        run: |
+          cd build
+          # https://github.com/emscripten-core/emscripten/issues/22264
+          sed 's/if(!stream.tty)/if(!stream.tty||!stream.tty.ops)/g' ledger.js > ledger.js.ioctl_fix
+          sed 's/,tty:true,/,tty:false,/g' ledger.js.ioctl_fix > ledger.js.fs_fix
+          # Enable testing from python scripts.
+          echo '#!'"$(which node)" > ledger.js
+          cat ledger.js.fs_fix >> ledger.js
```

It should be possible to do this without temporary files:
```suggestion
          sed -i '' \
          -e "s/^var/$(which node)\n&/" \
          -e 's/if(!stream.tty)/if(!stream.tty||!stream.tty.ops)/g' \
          -e 's/,tty:true,/,tty:false,/g' \
          ledger.js
```

---

Thanks! I modified it a bit to make it (hopefully) work with both BSD Sed and GNU Sed. Can you try it out?